service/history/historyEngine.go

```diff
@@ -578,6 +580,48 @@ func (e *historyEngineImpl) GetMutableState(
 	return response, nil
 }
 
+func (e *historyEngineImpl) QueryWorkflow(
+	ctx ctx.Context,
+	request *h.QueryWorkflowRequest,
+) (*h.QueryWorkflowResponse, error) {
+	context, release, retError := e.historyCache.getOrCreateWorkflowExecution(ctx, request.GetDomainUUID(), *request.GetExecution())
```

Comment:
can just use err here. do not really need `retError` and `err`, as you are returning `retError` on failure.